chirp/drivers/kenwood_d7.py

```diff
@@ -13,332 +13,3404 @@
+                 ("APO", "Automatic Power Off (APO)"))),
+             ('DTMF',
+                (("dtmfmem 00", "Number Store #0"),
+                 ("dtmfmem 01", "Number Store #1"),
+                 ("dtmfmem 02", "Number Store #2"),
+                 ("dtmfmem 03", "Number Store #3"),
+                 ("dtmfmem 04", "Number Store #4"),
+                 ("dtmfmem 05", "Number Store #5"),
+                 ("dtmfmem 06", "Number Store #6"),
+                 ("dtmfmem 07", "Number Store #7"),
+                 ("dtmfmem 08", "Number Store #8"),
+                 ("dtmfmem 09", "Number Store #9"),
+                 ("TSP", "TX Speed"),
+                 ("TXH", "TX Hold"),
+                 ("PT", "Pause"))),
+             ('TNC',
+                (("DTB", "Data band select"),
+                 ("DS", "DCD sense"))),
+             ('AUX',
+                (("ARO", "Automatic Repeater Offset"),
+                 ("SCR", "Scan Resume"),
+                 ("BEP", "Key Beep"),
+                 ("ELK", "Tuning Enable"),
+                 ("TXS", "TX Inhibit"),
+                 ("AIP", "Advanced Intercept Point"),
+                 ("TH", "TX Hold, 1750 Hz"),
+                 ("XXX", "VHF band narrow TX deviation"))))),
+        ('APRS',
+            (("MYC", "My call sign"),
+             ("GU", "GPS receiver"),
+             # Untested... D7G and D700 take an index, but since D7A only has
+             # one position and is older, it may not.
+             # ("MP", "Latitude / longitude data"),
+             ("POSC", "Position comment"),
+             ("ICO", "Station icon"),
+             ("STAT", "Status text"),
+             ("TXI", "Beacon transmit interval"),
+             ("PP", "Packet path"),
+             ("DTX", "Beacon transmit method"),
+             ("UPR", "Group code"),
+             ("ARL", "Reception restriction distance"),
+             ("TEMP", "Unit for temperature"))),
+        ('SSTV',
+            (("SMY", "My call sign"),
+             ("MAC", "Color for call sign"),
+             ("SMSG", "Message"),
+             ("SMC", "Color for message"),
+             ("RSV", "RSV report"),
+             ("RSC", "Color for RSV report"),
+             ("VCS", "VC-H1 Control"))),
+        ('SkyCommand',
+            (("SCC", "Commander call sign"),
+             ("SCT", "Transporter call sign"),
+             ("SKTN", "Tone frequency select"))))
 
 
 @directory.register
-class THD7GRadio(THD7Radio):
+class THD7GRadio(KenwoodD7Family):
     """Kenwood TH-D7G"""
     MODEL = "TH-D7G"
 
-    _SETTINGS_OPTIONS = {
-        "BEPT": ["Off", "Mine", "All New", "All"],  # Added "All"
-    }
+    _SETTINGS = {
+        "DS": {'type': 'list',
+               'values': ("Data Band", "Both Bands", "Ignore DCD")},
+        "DSPA": {'type': 'list',
+                 'values': ("Entire Display", "One Line")},
+        "ICO": {'type': 'map',
+                'map': (('Kenwood', '0,0'),
+                        ('Runner', '0,1'),
```

Comment:
Don't you want to re-use as much of this as you can from the sibling?

Reply:
I do, but I don't actually know how to do it well... the only difference between the D7G and the D700 is a single extra entry in the D700.  I'm not too worried about the D7 since it doesn't have very many options.

Basically, I'm not sure how/when storage is allocated for these, and I'm not sure how to grab class values from other classes.

Reply:
It's not really a big deal, it's just a lot of duplication. I think you should be able to do something like:
```
'map': KenwoodD7._SETTINGS['ICO']['map'] + (('WEATHER Station (blue)', '1,/#'), ...)
```

Reply:
Done